src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -741,11 +765,107 @@ InsertSelectQuerySupported(Query *queryTree, RangeTblEntry *insertRte,
 		}
 	}
 
+
+	if (!AllRelationRestrictionsContainUninstantiatedQual(restrictionContext))
+	{
+		return DeferredError(ERRCODE_FEATURE_NOT_SUPPORTED,
+							 "cannot plan distributed query since all join conditions in the query "
```

Comment:
I'd like to think a bit more on the wording here. But, I'm open to any suggestions